inst/tutorials/r_and_rstudio_basic/r_and_rstudio_basic.Rmd

```diff
@@ -0,0 +1,284 @@
+
+For example, we can load our Saanich data into R with `read_csv` for a comma-separated file and specify the arguments that describe our data as follows:
+
+- `file`: the name of the file you want to load
+- `col_names`: can take either the value `TRUE` or `FALSE` and tells R if the first row contains column names
+
+```{r, eval = FALSE}
+read_csv(file="data/Saanich_OTU_metadata.csv", col_names = TRUE)
+```
+
+```{r, echo = FALSE}
+OTU_metadata_table <- combined
+```
+
+### Save data in the environment
+
+Since we want to do more with our data after reading it in, we need to save it as a variable in R as we did previously with the `<-` operator. You can choose to name the object whatever you like, though this module assumes the names used below.
+
+```{r eval = FALSE}
+OTU_metadata_table <- read_csv(file="data/Saanich_OTU_metadata.csv", col_names = TRUE)
+```
+
+### Data exploration
+
+Let's explore the data that we've imported into R.
+
+Using different functions, we can look at the dimensions of our data, number of rows, and number of columns:
+```{r}
+#number of rows followed by number of columns
+dim(OTU_metadata_table)
+
+#number of rows
+nrow(OTU_metadata_table)
+
+#number of columns
+ncol(OTU_metadata_table)
+```
+
+We can list the column names using `colnames()`:
+```{r}
+colnames(OTU_metadata_table)
+```
+
+We can select to work with only specific columns/variables from our table using the `select()` function:
+```{r}
+restricted_columns <- select(OTU_metadata_table, Depth, OTU0001, OTU0002,
+                             OTU0004)
+```
+
+We can filter for specific rows based on a column value using the `filter()` function. Here we restrict for rows where the value for `Depth` is smaller than 135. 
+
+```{r}
+above_135_depth <- filter(OTU_metadata_table, Depth < 135)
+```
+
+We can also only choose to work with specific rows based on their position in our data table using the `slice()` function. 
+
+```{r}
+first_five_rows <- slice(OTU_metadata_table, 1:5)
+```
+
+
+
+## Getting Help
+You can get help with any function in R by inputting `?function_name` into the Console. This will open a window in the bottom right under the Help tab with information on that function, including input options and example code.
+
+```{r eval = FALSE}
+?read_delim
+```
+
+The **Description** section tells us that `read_delim()` is a general case of the function we used, `read_csv()`, and `read_tsv()`.  
+
+The **Usage** section tells us the inputs that need to be specified and default inputs of read_delim:
+
+- `file` and `delim` need to be specified as they are not followed by `=`
+- all other parameters have a default value e.g. `quote = "\"` and do not have to be specified to run the function.
+
+The **Arguments** Section describes the requirements of each input argument in detail.   
+
+The **Examples** Section has examples of the function that can be directly copy and pasted into your terminal and ran. 
+
+
```

Comment:
Another example from base R that may be widely used is the function `nrow()` 

```{r eval = FALSE}
?nrow
```

The **Description** section tells us that `nrow()` from a matrix or an array.

The **Usage** section tells us the inputs that need to be specified and default inputs of `nrow()`:

- `x` is the data matrix or array for which the user is interested in identifying the number of rows

The **Arguments** Section describes the requirements of each input argument in detail.

The **Examples** Section has examples of the function that can be directly copy and pasted into your terminal and ran.



Tidyverse is a wrapper for many valuable functions widely used in R. One of the examples from Tidyverse would be `select()`

```{r eval = FALSE}
if (!require("tidyverse")) install.packages("tidyverse")
library(tidyverse)
?select
```

The **Description** section tells us that `select()` can be used to select certain columns from a parent dataset, or optionally rename the columns

The **Overview of selection features** section provides the user a list of operators and selection helpers to fully realize the power of `select()` 

The **Usage** section tells us the inputs that need to be specified and default inputs of `select()`

- `.data` is a mandatory input which refers to the parent dataset or tibble to subset from
- the helper functions and list of operators could be used to specify the columns to subset

The **Value** section describes an object integral to `select()`. A more descriptive account can be read in the help section

The **Method** section describes the implementation method of the function `select()`

The **Examples** Section has examples of the function that can be directly copy and pasted into your terminal and ran.